### PR TITLE
rc/1.75.0

### PR DESCRIPTION
**Pull request by ciaranschutte**
**Description of changes**

#896 Scope Mongo Container Version
#895 Enable Debug Logs for Integration Tests
#891 Use SWC + Upgrade TS-Node
#893 Update Swagger
#897 Remove SWC
#849 Validation Rules for Program Exceptions
#884 Persistent Post Submission Errors
#890 Improved error handling for exceptions

+
#901 Addresses TS casting issues
#902 Some documentation
#903 Dependencies versions bump

**Type of Change**

- [ ] Bug
- [ ] Refactor
- [ ] New Feature
- [x] Release Candidate

**Checklist before requesting review:**

- [ ] Check branch (code change PRs go to `develop` not master)
- [ ] Check copyrights for new files
- [ ] Manual testing
- [ ] Regression tests completed and passing (double check number of tests).
- [ ] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)
- [ ] Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

**Review by joneubank — CHANGES_REQUESTED** [review]
@demariadaniel theres a couple places where the type casting I think is concerning. On review, I don't think this will break when run but I would like to clean this up before we merge to main. My apologies if I previously overlooked this during a review only to raise the issue now - I guess that's the value in having multiple steps to check during deployment.

File: src/clinical/service-worker-thread/tasks.ts [hunk 1]
```diff
@@ -100,21 +100,22 @@ const isEntityInQuery = (entityName: ClinicalEntitySchemaNames, entityTypes: str
...
-  const currentDonor = currentRecord['donor_id'];
-  const nextDonor = nextRecord['donor_id'];
 
   // -1 Current lower index than Next, +1 Current higher index than Next
   let order = 0;
   const isDescending = sort.includes('-') ? -1 : 1;
 
   if (isDefaultSort) {
+    const { donorId: currentDonorId } = currentRecord as Donor;
+    const { donorId: nextDonor } = nextRecord as Donor;
```
[the remark this answers is not in the document]
Comment (demariadaniel) [reply src/clinical/service-worker-thread/tasks.ts:114]: @joneubank @ciaranschutte I've updated Types related to Submitted Data in https://github.com/icgc-argo/argo-clinical/pull/901

File: src/clinical/service-worker-thread/tasks.ts [hunk 2]
```diff
@@ -124,18 +125,22 @@ const sortDocs = (sort: string, entityName: string, completionStats: CompletionR
     const completionSort = completionA === completionB ? 0 : completionA > completionB ? -1 : 1;
     order = completionSort;
   } else {
+    const currentEntry = currentRecord as ClinicalInfo;
+    const nextEntry = nextRecord as ClinicalInfo;
```
[the remark this answers is not in the document]
Comment (demariadaniel) [reply src/clinical/service-worker-thread/tasks.ts:129]: Updated in #901

File: src/clinical/clinical-service.ts [hunk 3]
```diff
@@ -240,19 +253,88 @@ export const getClinicalEntityMigrationErrors = async (programId: string, query:
...
+        errors.forEach(entityErrorObject => {
+          const currentEntityErrorData: [
+            string | EntityAlias,
+            readonly DeepReadonly<SchemaValidationError>[],
+          ] = Object.entries(entityErrorObject)[0];
+
+          const entityName = currentEntityErrorData[0] as EntityAlias;
+          const entityErrors = currentEntityErrorData[1] as readonly DeepReadonly<
+            SchemaValidationError
+          >[];
```
[the remark this answers is not in the document]
Comment (demariadaniel) [reply src/clinical/clinical-service.ts:270]: Updated in #901

**Review by joneubank — APPROVED** [review]

**Pull request merged**